lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -1284,6 +1294,38 @@ prop_MultiPoolCertificateSequence_coverage mpcs = checkCoverage
     certificateSequences = getSinglePoolSequences mpcs
     poolCount = length certificateSequences
 
+
+-- | read . put == pure
+prop_putHeaderListHeader
+    :: DBLayer IO
+    -> [BlockHeader]
+    -> Positive Int
+    -> Property
+prop_putHeaderListHeader DBLayer{..} headers (Positive k) =
+    monadicIO (setup >> prop)
+  where
+    sortedHeaders = sortBy (\a b -> compare (blockHeight a) (blockHeight b))
+        . L.nubBy (\a b -> (blockHeight a) == (blockHeight b)) . nub $ headers
+    expected = reverse . (if k == 0 then id else take k) . reverse $ sortedHeaders
+    setup = run $ atomically cleanDB
+    prop = do
+        void $ run $ atomically $ forM sortedHeaders putHeader
+        lHeaders <- run $ atomically $ listHeaders k
+        monitor $ counterexample $ unlines
+            [ "Read from DB: " <> show lHeaders
+            , "Expected: " <> show expected
+            , "Input: " <> show sortedHeaders
+            , "k: " <> show k]
+        monitor $ classify (null lHeaders) "Empty distributions"
+        assert (lHeaders == expected)
+        assert (strictlyIncreasing . fmap blockHeight $ lHeaders)
```

Comment:
**Suggestion**: use `assertWith` here, since we have more than one assertion.

**Justification**: when we encounter a failure, it's good to know _which_ assertion failed.

So something like:

```suggestion
        assertWith "read back the headers we expected"
            (lHeaders == expected)
        assertWith "headers are increasing in block height"
            (strictlyIncreasing . fmap blockHeight $ lHeaders)
```